packages/typespec-azure-resource-manager/src/private.decorators.ts

```diff
@@ -48,6 +48,31 @@ export function $omitIfEmpty(context: DecoratorContext, entity: Model, propertyN
   }
 }
 
+export function $enforceConstraint(
+  context: DecoratorContext,
+  entity: Operation | Model,
+  sourceType: Model,
```

Comment:
See above, might want to add an 'action message'  argument telling  the user what action to take to correct the error.